### DISCUSSION TITLE
How do I delete a device that is no longer in my home but still reported by the integration?

### DISCUSSION THREAD

**Discussion by ptr727** [RobertD502/home-assistant-flair#84] [Q&A] [answered]
I replaced three puck gateways with three bridge gateways.  
The pucks are no longer in the flair app, but they are still present in the HA integration, but reports as unavailable.  
When I look at the puck device in HA I cannot find any way to delete it from the flair integration.

How do I delete the device from the integration when it no longer exists?

**Comment by RobertD502** [comment]
1. Open the `__init__.py` file in `custom_components/flair`.
2. Under the line `from homeassistant.core import HomeAssistant` add this new line: `from homeassistant.helpers import device_registry as dr`
3. At the very end of the file, add the following function:

```python
async def async_remove_config_entry_device(
    hass: HomeAssistant, config_entry: ConfigEntry, device_entry: dr.DeviceEntry
) -> bool:
    """Remove a config entry from a device."""
    return True
```

Save the changes and restart Home Assistant. You should now see a delete button when you click on the device.

These changes will get added in the next release, but you can go ahead and make them yourself to get rid of the devices that are no longer on your Flair account.

**Reply by ptr727** [reply]
Ok, I'll give that a go tomorrow when I have a bit more time, thank you.

**Reply by RobertD502** [reply, marked as answer]
`0.2.2` has been released, which includes the ability to delete devices from the integration.

**Reply by ptr727** [reply]
Thank you, that worked.
